src/CoreWCF.Primitives/src/CoreWCF/Security/SecuritySessionSecurityTokenAuthenticator.cs

```diff
@@ -0,0 +1,1366 @@
+                using (XmlDictionaryReader bodyReader = request.GetReaderAtBodyContents())
+                {
+                    rst = this.StandardsManager.TrustDriver.CreateRequestSecurityToken(bodyReader);
+                    request.ReadFromBodyContentsToEnd(bodyReader);
+                }
+                if (rst.RequestType != null && rst.RequestType != this.StandardsManager.TrustDriver.RequestTypeIssue)
+                {
+                    throw TraceUtility.ThrowHelperWarning(new SecurityNegotiationException(SR.Format(SR.InvalidRstRequestType, rst.RequestType)), request);
+                }
+                // echo the AppliesTo in the reply if it is an issue request
+                EndpointAddress appliesTo = null;
+                DataContractSerializer appliesToSerializer;
+                string appliesToName;
+                string appliesToNamespace;
+                rst.GetAppliesToQName(out appliesToName, out appliesToNamespace);
+                if (appliesToName == AddressingStrings.EndpointReference && appliesToNamespace == request.Version.Addressing.Namespace)
+                {
+                    throw new NotImplementedException();
+                    //TODO need to check
+                    /* if (request.Version.Addressing == AddressingVersion.WSAddressing10)
+                     {
+                         appliesToSerializer = DataContractSerializerDefaults.CreateSerializer(typeof(EndpointAddress10), DataContractSerializerDefaults.MaxItemsInObjectGraph);
+                         appliesTo = rst.GetAppliesTo<EndpointAddress10>(appliesToSerializer).ToEndpointAddress();
+                     }
+                     else if (request.Version.Addressing == AddressingVersion.WSAddressingAugust2004)
+                     {
+                         appliesToSerializer = DataContractSerializerDefaults.CreateSerializer(typeof(EndpointAddressAugust2004), DataContractSerializerDefaults.MaxItemsInObjectGraph);
+                         appliesTo = rst.GetAppliesTo<EndpointAddressAugust2004>(appliesToSerializer).ToEndpointAddress();
+                     }
+                     else
+                     {
+                         throw DiagnosticUtility.ExceptionUtility.ThrowHelperError(
+                             new ProtocolException(SR.Format(SR.AddressingVersionNotSupported, request.Version.Addressing)));
+                     }*/
+                }
+                else
+                {
+                    appliesTo = null;
+                    appliesToSerializer = null;
+                }
+                if (this.shouldMatchRstWithEndpointFilter)
+                {
+                    SecurityUtils.MatchRstWithEndpointFilter(request, this.endpointFilterTable, this.listenUri);
+                }
+                RequestSecurityTokenResponse rstr;
+                SecurityContextSecurityToken issuedToken = this.IssueToken(rst, request, null, null, out rstr);
+                //if (appliesTo != null)
```

Comment:
This also needs to be there. Is there a reason why you chose to disable this feature?